### PR TITLE
Github basic integration finished, only left fixing bugs and UI

### PR DESCRIPTION
**Pull request by franciscorubin**

### CODE REVIEW

File: app/src/components/core/connectors/github/#githubConnector.service.ts#~ [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    import IHttpService = angular.IHttpService;
+    import IIntervalService = angular.IIntervalService;
+    import IWindowService = angular.IWindowService;
+    import IPromise = angular.IPromise;
+
+    export class GithubConnector {
+
+        static $inject = ['$http', '$window', '$interval'];
+        private clientId = "e259d761d8d0805e29ad";
+        private clientSecret = "d34f0f9034dfdf25739e12c3f1861d084d5021ff";
```
Comment (edgarmueller) [inline app/src/components/core/connectors/github/#githubConnector.service.ts#~:12]: Aren't we supposed to keep this a secret?

File: app/src/components/core/connectors/github/#githubConnector.service.ts#~ [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+            var error = url.match(/[&\?]error=([^&]+)/);
+            if (error) return null;
+            var match = url.match(/[&\?]code=([\w\/\-]+)/)[1];
+            return match;
+        }
+
+        getToken(code:string, callback):void{
+            this.$http.post("https://github.com/login/oauth/access_token", {
+	        "headers": {
+		    "Access-Control-Allow-Origin":"*"
```
Comment (edgarmueller) [inline app/src/components/core/connectors/github/#githubConnector.service.ts#~:46]: Fix indentation.

File: app/src/components/core/connectors/github/#githubConnector.service.ts#~ [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+		},
+		"data": {
+		                "code":code,
+                "client_id": this.clientId,
+                "client_secret": this.clientSecret
+		}
+            }).success((response:any) => {
+	    console.log(response);
```
Comment (edgarmueller) [inline app/src/components/core/connectors/github/#githubConnector.service.ts#~:54]: Here as well (best to re-check for all).

File: app/src/components/core/connectors/github/githubConnector.service.ts [hunk 4]
```diff
@@ -1,19 +1,124 @@
...
 
+        private repoList;
+
+        private fileLoaders: FileLoader[];
+	private fileLevel: GithubFileLevel;
+        constructor(private $http:IHttpService, private $window:IWindowService, private $q:IQService, private $location:ILocationService) {
+            this.url = $location.protocol() + "://" + $location.host() + ":" + $location.port();
+	    this.fileLoaders = [];
+
+	    this.addFileLoader();
```
Comment (edgarmueller) [inline app/src/components/core/connectors/github/githubConnector.service.ts]: Why do we need to call this twice?

File: app/src/components/core/connectors/github/githubConnector.service.ts [hunk 5]
```diff
@@ -1,19 +1,124 @@
...
+            if (file.getType() !== 'tree') {
+                throw new Error('Calling method "goIntoFolder" with a regular file instead of a folder!');
+            }
+            // Here I'm reusing previously loaded fileLevels
+            if (this.fileLevel) {
+                var child = this.fileLevel.getChild(file.getName());
+                if (child) {
+                    this.fileLevel = child;
+
+                    var deferred = this.$q.defer();
```
Comment (edgarmueller) [inline app/src/components/core/connectors/github/githubConnector.service.ts]: I think you can use `$q.when` here to return a promise of the `child`.

File: app/src/components/core/connectors/github/#githubConnector.service.ts#~ [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+                callback(response);
+            }).error((error:any) => {
+                console.log(error);
```
Comment (edgarmueller) [inline app/src/components/core/connectors/github/#githubConnector.service.ts#~:57]: Re-check whether there are obsolete console.log(s).

File: app/src/components/core/connectors/github/model/fileLevel.ts~ [hunk 7]
```diff
@@ -0,0 +1,13 @@
+module app.core.connectors{
+       export class GithubFileLevel{
+       	      private files[]: GithubFile[];
+       	      constructor(initialData: Array<any>, private parentLevel: GithubFileLevel){
+		  this.files = initialData.map(function (obj) {
```
Comment (edgarmueller) [inline app/src/components/core/connectors/github/model/fileLevel.ts~:5]: use fat-arrow syntax: `obj => new GithubFile(obj)`

File: app/src/components/core/dataschema/dataschema.service.ts [hunk 8]
```diff
@@ -127,7 +127,7 @@ module app.core.dataschema {
         removeProperty(name:string, path:string[]):boolean {
             var parent = this.getFolderAt(path);
 
-            if (parent === null || !parent.hasOwnProperty('properties') || !parent.properties.hasOwnProperty(name)) {
+            if (parent === null || !parent.hasOwnProperty('properties') || !parent.properties.hasOwnProperty(name.split('/').pop())) {
```
Comment (edgarmueller) [inline app/src/components/core/dataschema/dataschema.service.ts]: I would extract the last conditional into a private predicate method. It's not obvious to me what `!parent.properties.hasOwnProperty(name.split('/').pop()))` does.

File: app/src/components/dialogs/abstract/wizard/abstractWizard.ts [hunk 9]
```diff
@@ -50,9 +72,13 @@ module app.dialogs {
...
 
         next():void {
-            if (this.hasNext()) {
+	
+          if (this.hasNext()) {
+	        console.log('hasNext');
                 if (this.currentStep().shallSubmit()) {
+		    console.log('shallsubmit');
                     this.currentStep().submit().then(() => {
+		        console.log('submittedddddddd');
```
Comment (edgarmueller) [inline app/src/components/dialogs/abstract/wizard/abstractWizard.ts]: Make sure to check for console.log :) (See comment above).

File: app/src/components/dialogs/dataschemaImport/githubHook/githubHookRepoStep.controller.ts [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+        }
+
+        getBranches(): any{
+            return this.branches;
+        }
+
+        selectBranch(branch): void{
+            this.selectedBranch = branch;
+
+            //this.wizard.next();
```
Comment (edgarmueller) [inline app/src/components/dialogs/dataschemaImport/githubHook/githubHookRepoStep.controller.ts]: Remove?

File: app/src/components/dialogs/dataschemaImport/githubHook/githubHookRepoStep.controller.ts [hunk 11]
```diff
@@ -0,0 +1,72 @@
+module app.dialogs.dataschemaimport {
+
+    import SocioCortexConnector = app.core.connectors.SocioCortexConnector;
```
Comment (edgarmueller) [inline app/src/components/dialogs/dataschemaImport/githubHook/githubHookRepoStep.controller.ts]: Where do we use the SocioCortexConntector?

File: app/src/components/dialogs/dataschemaImport/githubHook/githubHookUISchemaStep.controller.ts [hunk 12]
```diff
@@ -0,0 +1,92 @@
...
+        goToParentFolder():void{
+            this.githubConnector.goToParentFolder();
+        }
+
+        submit():IPromise<any> {
+            if(this.gonnaSkip){
+                var result = {
+                    dataSchema: this.githubConnector.getFileLoader(0).loadedFileContents
+                }
+                var deferred = this.$q.defer();
```
Comment (edgarmueller) [inline app/src/components/dialogs/dataschemaImport/githubHook/githubHookUISchemaStep.controller.ts]: Again, `$q.when` should be also applicable here.

File: app/src/components/dialogs/dataschemaImport/githubHook/githubHookUISchemaStep.controller.ts [hunk 13]
```diff
@@ -0,0 +1,92 @@
...
+                deferred.resolve(result);
+                return deferred.promise;
+            }
+            return this.githubConnector.loadFile(this.selectedFile, this.fileSelectorID).then((res)=> {
+                var result = {
+                    dataSchema: this.githubConnector.getFileLoader(0).loadedFileContents,
+                    uiSchema: res
+                }
+		return result;
```
Comment (edgarmueller) [inline app/src/components/dialogs/dataschemaImport/githubHook/githubHookUISchemaStep.controller.ts]: Inline `return`

File: backend/config/keys.js [hunk 14]
```diff
@@ -0,0 +1,12 @@
+/**
+ * Created by pancho111203 on 15/02/16.
+ */
+
+var keys = {};
+
+keys.clientId = "5546933a86952d67f3c0";
+keys.clientSecret = "71a524daac2a72d20a94a28ae13032f77ea19170";
```
Comment (edgarmueller) [inline backend/config/keys.js]: Are we supposed to share this?

**Comment by edgarmueller** [comment]
Great work so far! Maybe we can go over the remaining issue together later.

File: backend/config/keys.js (hunk 14 above)
Comment (franciscorubin) [inline backend/config/keys.js]: We should hide the final keys from github(this ones are temporary). I don't know what is the best way, maybe environment variables?

**Pull request merged**